internal/provider/inventory_data_source.go

```diff
@@ -5,102 +5,131 @@ import (
 func NewInventoryDataSource() datasource.DataSource {
-	return &inventoryDataSource{}
+	return &InventoryDataSource{}
 }
 
 // inventoryDataSource is the data source implementation.
-type inventoryDataSource struct {
+type InventoryDataSource struct {
 	client *AAPClient
 }
 
 // Metadata returns the data source type name.
-func (d *inventoryDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+func (d *InventoryDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
 	resp.TypeName = req.ProviderTypeName + "_inventory"
 }
 
 // Schema defines the schema for the data source.
-func (d *inventoryDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+func (d *InventoryDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
 	resp.Schema = schema.Schema{
 		Attributes: map[string]schema.Attribute{
 			"id": schema.Int64Attribute{
 				Required: true,
 			},
-			"groups": schema.MapNestedAttribute{
-				NestedObject: schema.NestedAttributeObject{
-					Attributes: map[string]schema.Attribute{
-						"hosts": schema.ListAttribute{
-							ElementType: types.StringType,
-							Computed:    true,
-						},
-						"children": schema.ListAttribute{
-							ElementType: types.StringType,
-							Computed:    true,
-						},
-					},
-				},
+			"organization": schema.Int64Attribute{
 				Computed: true,
+				Description: "Identifier for the organization the inventory should be created in. " +
+					"If not provided, the inventory will be created in the default organization.",
 			},
-			"hosts": schema.MapNestedAttribute{
-				NestedObject: schema.NestedAttributeObject{
-					Attributes: map[string]schema.Attribute{
-						"hostvars": schema.MapAttribute{
-							ElementType: types.StringType,
-							Computed:    true,
-						},
-					},
-				},
+			"url": schema.StringAttribute{
 				Computed: true,
 			},
+			"name": schema.StringAttribute{
+				Computed: true,
+			},
+			"description": schema.StringAttribute{
+				Computed: true,
+			},
+			"variables": schema.StringAttribute{
+				Optional:   true,
+				CustomType: jsontypes.NormalizedType{},
+			},
 		},
 	}
 }
 
-const ungroupedName string = "ungrouped"
-const allgroupsName string = "all"
+// inventoryDataSourceModel maps the data source schema data.
+type InventoryDataSourceModel struct {
+	Id           types.Int64          `tfsdk:"id"`
+	Organization types.Int64          `tfsdk:"organization"`
+	Url          types.String         `tfsdk:"url"`
+	Name         types.String         `tfsdk:"name"`
+	Description  types.String         `tfsdk:"description"`
+	Variables    jsontypes.Normalized `tfsdk:"variables"`
+}
+
+func (d *InventoryDataSource) ReadInventory(id string) (*InventoryDataSourceModel, error) {
+	resp, body, err := d.client.doRequest("GET", "api/v2/inventories/"+id+"/", nil)
```

Comment:
Please take this opportunity to refactor all code-performing API requests to be aligned with what has been done on the other resources. Here is an example https://github.com/ansible/terraform-provider-aap/blob/d53ac596953fd48d26aa087375b0fbf5c04a6f8b/internal/provider/host_resource.go#L218-L229